Objects/rangeobject.c

```diff
@@ -766,27 +766,55 @@ PyTypeObject PyRange_Type = {
 
 typedef struct {
         PyObject_HEAD
-        long    index;
         long    start;
+        long    stop;
         long    step;
-        long    len;
 } rangeiterobject;
 
+/* Return number of items in range (lo, hi, step).  step != 0
+ * required.  The result always fits in an unsigned long.
+ */
+static unsigned long
+get_len_of_range(long lo, long hi, long step)
```

Comment:
It would be a smaller change to just put a forward declaration here instead of moving the entire function around.

Reply:
Yes, but in long term it is better location for this function. It does not make the change too large, I think.

I you prefer, I'll return it back and put a forward declaration.